scriptworker/test/test_client.py

```diff
@@ -232,7 +243,10 @@ def test_sync_main_runs_fully(config, event_loop, should_validate_task):
         kwargs['config_path'] = f.name
         client.sync_main(async_main, **kwargs)
 
-    assert next(generator) == 1 # async_main was called once
+    for i in count:
+        await i  # suppress coroutine not awaited warning
+    assert len(count) == 1  # run_until_complete was called once
+    assert next(generator) == 1  # async_main was called once
```

Comment:
Nit: `count` and `generator` seem to implement the same intent (counting the number of times a function was called) but:
* they do it on 2 different functions
* they're implemented differently.

In order to make the reading easy, I'd suggest to append the mocked function name on the variable name (eg: `count_on_run_until_complete`).
As a matter of consistency, let's make these 2 counters implemented the same way. I have no preference whether to use a generator or a list. 

What do you think?

Reply:
Sure. I'm going to use a list because I need to await the saved coroutines.